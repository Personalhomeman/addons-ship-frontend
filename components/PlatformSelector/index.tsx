import { Base, Flex, Icon, TypeColors } from '@bitrise/bitkit';
import cx from 'classnames';

import { Platform } from '@/models';

import css from './style.scss';

interface PlatformSelectorProps {
  platform: Platform;
  onClick: (platform: Platform) => void;
}

export default ({ platform, onClick }: PlatformSelectorProps) => {
  const color = (iosStyle: TypeColors, androidStyle: TypeColors) => (platform === 'ios' ? iosStyle : androidStyle);
  return (
    <Flex alignChildren="middle" direction="horizontal">
      <Flex
        direction="horizontal"
        className={css.platform}
        alignChildren="middle"
        backgroundColor="grape-1"
        borderRadius="x2"
      >
        <Base
          className={cx(css.Base, { [css.selected]: platform === 'ios' })}
          paddingHorizontal="x6"
          paddingVertical="x3"
          onClick={() => onClick('ios')}
        >
          <Icon name="PlatformsApple" color={color('white', 'grape-3')} />
        </Base>
        <Base
          className={cx(css.Base, { [css.selected]: platform === 'android' })}
          paddingHorizontal="x6"
          paddingVertical="x3"
          onClick={() => onClick('android')}
        >
          <Icon name="PlatformsAndroid" color={color('grape-3', 'white')} />
        </Base>
      </Flex>
    </Flex>
  );
};
