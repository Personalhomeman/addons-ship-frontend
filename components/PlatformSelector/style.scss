@import '~@/assets/style/mixins';
@import '~@/assets/style/colors';

.platform {
  border: 1px solid var(--color-eggplant);

  .Base {
    cursor: pointer;

    &:first-child {
      @include border-left-radius(8px);
    }
    &:last-child {
      @include border-right-radius(8px);
    }
    &.selected {
      background-color: var(--color-grape--3);
    }
  }
}
