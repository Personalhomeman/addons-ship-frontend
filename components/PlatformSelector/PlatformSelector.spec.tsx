import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import PlatformSelector from '.';

describe('PlatformSelector', () => {
  it('renders correctly', () => {
    const tree = toJSON(shallow(<PlatformSelector platform="ios" onClick={() => {}} />));
    expect(tree).toMatchSnapshot();
  });
});
