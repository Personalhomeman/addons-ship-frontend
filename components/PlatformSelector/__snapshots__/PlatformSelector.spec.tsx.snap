// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlatformSelector renders correctly 1`] = `
<Flex
  alignChildren="middle"
  direction="horizontal"
>
  <Flex
    alignChildren="middle"
    backgroundColor="grape-1"
    borderRadius="x2"
    className="platform"
    direction="horizontal"
  >
    <Base
      className="Base selected"
      onClick={[Function]}
      paddingHorizontal="x6"
      paddingVertical="x3"
    >
      <Icon
        color="white"
        name="PlatformsApple"
        size="1.5rem"
      />
    </Base>
    <Base
      className="Base"
      onClick={[Function]}
      paddingHorizontal="x6"
      paddingVertical="x3"
    >
      <Icon
        color="grape-3"
        name="PlatformsAndroid"
        size="1.5rem"
      />
    </Base>
  </Flex>
</Flex>
`;
